docs/rfcs/0020-rotate-hash-peppers.md

```diff
@@ -0,0 +1,379 @@
+5. Where the hash leak does become relevant, is if an attacker knows a patient MBI and the hashing algorithm; in that case they can generate a hashed MBI and pass that to search services that require a hashed MBI; that is if they can even connect to BFD (see item 2 above). However, even that scenario would fall into the _merely interesting_ scenario; if the attacker has access to log files, they could simply just scrape all log entries that have an MBI hash in the URL, and just replay those URL(s), again assuming they can get access to BFD services.
+6. How about the scenario where the _bad actor_ has access to both BFD log entries and BFD's hashing algorithm; using that knowlege, one could try a _brute force_ attack by controlled generation of an MBI (see https://www.cms.gov/medicare/new-medicare-card/understanding-the-mbi-with-format.pdf for the structure of an MBI) until the generated hash MBI matches one from the log file(s). While mathematically challenging to say the least, it is hypothetically possible.
+### So We Leaked the Hash Algorithm
+Prior to delving into possible solutions, and more germane to this RFC, some background on how BFD currently might handle a leak of the algorithm that both BFD and peering partners share.
+
+To start with, BFD has never encountered the situation where a leaked component of the hashing alogritm would require immediate mitigation; there have been some muted table-top and/or game-day exercises, but nothing concrete and for real. So what follows is some back-of-the-napkin tasks that would need to be completed in the current environment in such a scenario.
+- incident declared denoting hashing algorithm is compromised; depending on the severity of the leak (i.e., how long has it been compromised, etc.) may dictate the severity of the incident.
+- BFD services may not need to be taken offline; the hashing algorithm is used by a couple of BFD services; those service calls takes an MBI or HICN hash parameter; for example (cURL syntax):
+```
+identifier="https://bluebutton.cms.gov/resources/identifier/hicn-hash|6026b9c1f00ba3937a8876f909c1c0d806d36229111e030ab867a8fc91792851"
+or
+identifier="https://bluebutton.cms.gov/resources/identifier/mbi-hash|1425547895d49fdf40d4128ec03cfd53c6ca2a67af9f7adcc2845a00cf6a5af5"
+```
+- need to change components of the (leaked) hashing algorithm; the changes would most likely be constrained to the seed and/or iteration count. The new hash algorithm components would be persisted in the BFD SSM parameter store.
+- peering partners need to be alerted to the issue and pendinghashing  mitigation plan; changing the algorithm will require communicating the new hashing algorithm via an agreed upon secure channel. Since this affects all PPs, some amount of collaboration and synchronization of PPs will need to be implemented.
```

Comment:
```suggestion
- peering partners need to be alerted to the issue and pending hashing mitigation plan; changing the algorithm will require communicating the new hashing algorithm via an agreed upon secure channel. Since this affects all PPs, some amount of collaboration and synchronization of PPs will need to be implemented.
```

Reply:
fixed.